sky/data/storage.py

```diff
@@ -846,33 +962,14 @@ def from_metadata(cls, metadata: StorageMetadata,
 
         return storage_obj
 
-    def add_store(self,
-                  store_type: Union[str, StoreType],
-                  region: Optional[str] = None) -> AbstractStore:
-        """Initializes and adds a new store to the storage.
-
-        Invoked by the optimizer after it has selected a store to
-        add it to Storage.
-
-        Args:
-          store_type: StoreType; Type of the storage [S3, GCS, AZURE, R2, IBM]
-          region: str; Region to place the bucket in. Caller must ensure that
-            the region is valid for the chosen store_type.
-        """
-        if isinstance(store_type, str):
-            store_type = StoreType(store_type)
-
-        if store_type in self.stores:
-            if store_type == StoreType.AZURE:
-                azure_store_obj = self.stores[store_type]
-                assert isinstance(azure_store_obj, AzureBlobStore)
-                storage_account_name = azure_store_obj.storage_account_name
-                logger.info(f'Storage type {store_type} already exists under '
-                            f'storage account {storage_account_name!r}.')
-            else:
-                logger.info(f'Storage type {store_type} already exists.')
-            return self.stores[store_type]
-
+    def construct_store(
+        self,
+        store_type: StoreType,
+        region: Optional[str] = None,
+        storage_account_name: Optional[str] = None,
+        _allow_bucket_creation: bool = True  # pylint: disable=invalid-name
+    ) -> AbstractStore:
+        """Initialize store object and get/create bucket."""
```

Comment:
This new method seems conflicting with`add_store()`. When should a user use `construct_store()` over `add_store()`?  

Is this new method necessary? If we need to add a specific store at Storage initialization, we support that through the `stores` arg of `Storage`. Can we use that?

Reply:
I was worried that it could change the interface, but I get it now—it won't.